src/coreComponents/mesh/mpiCommunications/mpiCommunicationUtils.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file CommID.hpp
+ */
+
+#ifndef GEOSX_MESH_MPICOMMUNICATIONS_MPICOMMUNICATIONUTILS_HPP_
+#define GEOSX_MESH_MPICOMMUNICATIONS_MPICOMMUNICATIONUTILS_HPP_
+
+#include "common/DataTypes.hpp"
+
+namespace geosx
+{
+
+struct SyncFieldsID
+{
+  void addFields(FieldLocation const location, std::vector<string> const fieldNames )
+  {
+    string const key = generateKey(location);
+    addFields(fieldNames, key);
+  }
+
+  void addFields(FieldLocation const location, std::vector<string> const fieldNames, std::vector<string> const regionNames )
```

Comment:
I think the location is known already to be `elems`? We wouldn't ever use this for anything other than elements?

Reply:
yes, unless we start having regions also on other managers. I guess I can remove that but then change the function name to something like `addElementFields` ?

Reply:
Yeah...I think it is strange to take an argument, but only allow one valid value for that argument.